kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -75,14 +96,30 @@ public ScanBuilder getScanBuilder(Engine engine) {
     return new ScanBuilderImpl(dataPath, protocol, metadata, getSchema(engine), logReplay, engine);
   }
 
-  public Metadata getMetadata() {
-    return metadata;
+  ///////////////////
+  // Internal APIs //
```

Comment:
This comment seems misleading. Even if this line says "Internal", 3rd party developers may call the subsequent method as they are public.